src/main/java/net/minecraftforge/event/entity/living/LivingSpawnEvent.java

```diff
@@ -116,9 +141,27 @@ public CheckSpawn(EntityLiving entity, World world, float x, float y, float z)
     @Cancelable
     public static class SpecialSpawn extends LivingSpawnEvent
     {
+        private final @Nullable MobSpawnerBaseLogic spawner;
+
+        @Deprecated
         public SpecialSpawn(EntityLiving entity, World world, float x, float y, float z)
         {
             super(entity, world, x, y, z);
+            spawner = null;
+        }
+
+        /**
+         * @param spawner the position of a tileentity or approximate position of an entity that initiated the spawn if any
+         */
+        public SpecialSpawn(EntityLiving entity, World world, float x, float y, float z, @Nullable MobSpawnerBaseLogic spawner)
+        {
+            super(entity, world, x, y, z);
+            this.spawner = spawner;
+        }
+
+        public @Nullable MobSpawnerBaseLogic getSpawner()
```

Comment:
`@Nullable` should go on the line before, except in patches where size is being minimized.